operator/pkg/helmreconciler/reconciler.go

```diff
@@ -196,6 +196,20 @@ func (h *HelmReconciler) processRecursive(manifests name.ManifestMap) *v1alpha1.
 
 // Delete resources associated with the custom resource instance
 func (h *HelmReconciler) Delete() error {
+	// Delete IOP with revision:
+	// for this case we update the status field to pending if there are still proxies pointing to this revision
+	// and we do not prune shared resources, same effect as `istioctl uninstall --revision foo` command.
+	iop := h.iop
+	if iop.Spec != nil && iop.Spec.Revision != "" {
```

Comment:
spec==nil is treated as default. also nit: you can write this a bit more cleanly as:
if iop.Spec.Revision == "" {
  return h.Prune(nil, true)
}
...

Reply:
ok, updated the logic flow to be more clear. And just clarify, it called h.Prune(nil, true) in the original code for all cases. But in the new change, we want to call the prune by revision method only when revision is not empty